yarn/pom.xml

```diff
@@ -92,12 +92,28 @@
       <artifactId>jetty-servlet</artifactId>
     </dependency>
     <!-- End of shaded deps. -->
-    
+
+    <!--
+      SPARK-10059: Explicitly add JSP dependencies for tests since the MiniYARN cluster needs them.
+    -->
+    <dependency>
+      <groupId>org.eclipse.jetty.orbit</groupId>
+      <artifactId>javax.servlet.jsp</artifactId>
+      <version>2.2.0.v201112011158</version>
```

Comment:
Does this need to be synced with the servlet API version? Ideally I suppose it does but I don't know how hard that is to plumb. I thought we had servlet 3.x in the build but you have probably looked more recently. I know JSTL is versioned differently.